scheduler/src/cook/mesos/sandbox.clj

```diff
@@ -80,30 +80,16 @@
       sandbox-updater-publish-duration
       (doseq [task-id->sandbox-partition (partition-all batch-size task-id->sandbox)]
         (try
-          (let [datomic-db (d/db datomic-conn)
-                task-ids-with-sandbox (->> (d/q '[:find ?t
-                                                  :in $ [?t ...]
-                                                  :where
-                                                  [?e :instance/task-id ?t]
-                                                  [?e :instance/sandbox-directory ?s]]
-                                                datomic-db (keys task-id->sandbox-partition))
-                                           (map first)
-                                           (into #{}))]
-            (letfn [(task-id->instance-id [db task-id]
-                      (-> (d/entity db [:instance/task-id task-id])
-                          :db/id))
-                    (build-sandbox-txns [[task-id sandbox]]
-                      (when-not (contains? task-ids-with-sandbox task-id)
-                        (let [instance-id (task-id->instance-id datomic-db task-id)]
-                          [[:db/add instance-id :instance/sandbox-directory sandbox]])))]
-              (let [txns (mapcat build-sandbox-txns task-id->sandbox-partition)]
-                (when (seq txns)
-                  (log/info "Inserting" (count txns) "facts in sandbox state update")
-                  (meters/mark! sandbox-updater-tx-rate)
-                  (timers/time!
-                    sandbox-updater-tx-duration
-                    @(d/transact datomic-conn txns)))))
-            (send task-id->sandbox-agent clear-agent-state task-id->sandbox-partition))
+          (letfn [(build-sandbox-txns [[task-id sandbox]]
+                    [[:db/add [:instance/task-id task-id] :instance/sandbox-directory sandbox]])]
+            (let [txns (mapcat build-sandbox-txns task-id->sandbox-partition)]
```

Comment:
Nitpick: you can use `map` and then only one set of brackets in build-sandbox-txns